src/charm.py

```diff
@@ -104,18 +111,27 @@ def install_magma_access_gateway_snap() -> None:
             stdout=subprocess.PIPE,
         )
 
-    def install_magma_access_gateway(self) -> None:
+    def install_magma_access_gateway(self) -> int:
         """Installs Magma access gateway on the host.
 
         Returns:
-            None
+            Return code of the installation script
         """
         command = ["magma-access-gateway.install"]
         command.extend(self._install_arguments)
-        subprocess.run(
+        install_process = subprocess.run(
             command,
             stdout=subprocess.PIPE,
         )
+        logger.info(install_process.stdout)
+        return install_process.returncode
+
+    def reboot(self) -> None:
+        """Sends the command to reboot the machine in 1 minute."""
+        subprocess.run(
+            ["shutdown", "--reboot", "+1"],
```

Comment:
Why wait 1 min?

Reply:
If we reboot immediately, there is a chance that the charm execution will not complete, and Juju will interpret this as a failed hook. If we are too fast to reboot, it is also possible the status will not be updated properly. The `shutdown` command by default schedules the reboot in a minute, in this case, I added the argument to be more explicit about the intended behaviour.